arkouda/groupbyclass.py

```diff
@@ -970,6 +975,333 @@ def broadcast(self, values : pdarray, permute : bool=True) -> pdarray:
+
+            elif "categorical" in name:
+                from arkouda.categorical import Categorical
+                keys.append(Categorical.attach(name))
+                categorical = True
+
+            else:
+                raise RegistrationError(f"Unknown type associated with registered item: {user_defined_name}."
+                                        f" Supported types are: {groupable}")
+
+        if len(keys) == 0:
+            raise RegistrationError(f"Unable to attach to '{user_defined_name}' or '{user_defined_name}'"
+                                    f" is not registered")
+
+        perm_resp = generic_msg(cmd="attach", args=f"{user_defined_name}_categorical.keys.permutation") if\
+            categorical and len(keys) == 1 else generic_msg(cmd="attach", args=f"{user_defined_name}.permutation")
+
+        permutation = create_pdarray(perm_resp)
+        g = GroupBy.build_from_components(keys, permutation, user_defined_name)  # Call build_from_components method
+        return g
+
+    @staticmethod
+    @typechecked
+    def unregister_groupby_by_name(user_defined_name: str) -> None:
+        """
+        Function to unregister GroupBy object by name which was registered
+        with the arkouda server via register()
+
+        Parameters
+        ----------
+        user_defined_name : str
+            Name under which the GroupBy object was registered
+
+        Raises
+        -------
+        TypeError
+            if user_defined_name is not a string
+        RegistrationError
+            if there is an issue attempting to unregister any underlying components
+
+        See Also
+        --------
+        register, unregister, attach, is_registered
+        """
+        # We have 2 components, unregister both of them
+        from arkouda.categorical import Categorical
+        from re import match, compile
+        registry = list_registry()
+
+        regEx = compile(f"^{user_defined_name}_.+\\.keys$|^\\d+_{user_defined_name}_.+\\.keys$|"
+                        f"^(?:\\d+_)?{user_defined_name}_categorical\\.keys(?=\\.categories$)")
+        categorical = False
+        matches = 0
+        for name in registry:
+            # Search through registered items and find matches to the given name
+            x = match(regEx, name)
+            if x is not None:
+                matches += 1
+                # Only categorical requires a separate unregister case
+                if "categorical" in x.group():
+                    Categorical.unregister_categorical_by_name(x.group())
+                    categorical = True
+                else:
+                    unregister_pdarray_by_name(x.group())
+
+        # Only one permutation, so remove it after the loop
+        if matches == 1 and categorical:
+            unregister_pdarray_by_name(f"{user_defined_name}_categorical.keys.permutation")
+        else:
+            unregister_pdarray_by_name(f"{user_defined_name}.permutation")
```

Comment:
Again, because of the naming, this will not function properly in the case that it uses _categorical. The constructor will just rebuild the permutation.

Reply:
Comment above also explains this.
{user_defined_name}_categorical.keys is the "name" passed to the Categorical.register() method. Due to how values are registered, {user_defined_name}.permutation that is registered for the groupby will get overwritten in the registry when the GroupBy is built with a single Categorical as its key. This is why {user_defined_name}_categorical.keys.permutation is used in this case